cmd/release-controller/sync.go

```diff
@@ -514,6 +523,40 @@ func (c *Controller) syncReady(release *Release) error {
 	return nil
 }
 
+func (c *Controller) syncTerminal(release *Release) error {
+	terminalTags := sortedRawReleaseTags(release, releasePhaseAccepted, releasePhaseRejected)
+	if klog.V(5) && len(terminalTags) > 0 {
+		klog.Infof("terminal=%v", tagNames(terminalTags))
+	}
+	var tagAnnotations map[string]interface{}
+	for _, releaseTag := range terminalTags {
+		oldVerifyStatus := releaseTag.Annotations[releaseAnnotationVerify]
+		if len(oldVerifyStatus) == 0 {
+			klog.V(5).Infof("%s: Empty verify annotation, skipping", releaseTag.Name)
+			continue
+		}
+		status := make(VerificationStatusMap)
+		if err := json.Unmarshal([]byte(oldVerifyStatus), &status); err != nil {
+			klog.Errorf("Release %s has invalid verification status, ignoring: %v", releaseTag.Name, err)
+			continue
+		}
+		for i := range status {
+			status[i].TransitionTime = nil
+		}
+		newVerifyStatus := toJSONString(status)
```

Comment:
This is expensive and you are doing it on every single tag on every single loop.  In general anything in a sync loop has to be fast and efficient - if no change is needed, we don't do work.  What's concerning here is that once you've paid the cost once you still check it every time - the rest of the release controller is designed around the sync loop being relatively cheap to do, but here we're doing something relatively expensive.

Once transition time has been removed from all tags, this code is no longer necessary (the invariants elsewhere should cover it).  So effectively we are going to merge this code and then after a few hours we can completely remove this loop (since all data should be updated).  That removes the need to come up with a more complex structure here if the code were going to live a long time.

Please prepare a PR that builds on top of this change and removes the syncTerminal loop.

Reply:
https://github.com/openshift/release-controller/pull/208 - This PR has changes for both removing the transitionTime annotation and the syncTerminal loop